pkg/cose/cose.go

```diff
@@ -0,0 +1,183 @@
+package cose
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"math/big"
+
+	"github.com/blocky/nitrite/pkg/attestation"
+	"github.com/blocky/nitrite/pkg/nitrite_error"
+	"github.com/fxamacker/cbor/v2"
+)
+
+type CoseHeader struct {
+	Alg interface{} `cbor:"1,keyasint,omitempty" json:"alg,omitempty"`
+}
+
+type CosePayload struct {
+	_           struct{} `cbor:",toarray"`
+	Protected   []byte
+	Unprotected cbor.RawMessage
+	Payload     []byte
+	Signature   []byte
+}
+
+type CoseSignature struct {
+	_           struct{} `cbor:",toarray"`
+	Context     string
+	Protected   []byte
+	ExternalAAD []byte
+	Payload     []byte
+}
+
+func (h *CoseHeader) AlgorithmString() (string, bool) {
+	switch h.Alg.(type) {
+	case string:
+		return h.Alg.(string), true
+	}
+	return "", false
+}
+
+func (h *CoseHeader) AlgorithmInt() (int64, bool) {
```

Comment:
This function has no usages and is not tested. Do we need it?

Reply:
Good catch. removed.